nodejs/src/uws.js

```diff
@@ -432,33 +455,32 @@ class Server extends EventEmitter {
 
         native.server.group.onConnection(this.serverGroup, (external) => {
             const webSocket = new WebSocket(external);
+
             native.setUserData(external, webSocket);
             this._upgradeCallback(webSocket);
             _upgradeReq = null;
         });
 
-        if (options.port) {
-            if (options.host) {
-                this.httpServer.listen(options.port, options.host, ()=> {
-                      this.emit('listening');
-                      callback && callback();
-                });
-            } else {
-                this.httpServer.listen(options.port,  ()=> {
-                      this.emit('listening');
-                      callback && callback();
-                });
-            }
+        if (options.host) {
+            this.httpServer.listen(options.port, options.host, () => {
+                this.emit('listening');
+                callback && callback();
+            });
+        } else {
+            this.httpServer.listen(options.port,  () => {
```

Comment:
If you pass server as option, no host and no port, then it will still listen. It shouldn't listen if you pass it a server. This line also still has double space ",spacespace() => {".